### PR TITLE
Backport of docs: update redirected links into release/1.14.x

### PR DESCRIPTION
**Pull request by hc-github-team-consul-core**
## Backport

This PR is auto-generated from #16179 to be assessed for backporting due to the inclusion of the label type/docs-cherrypick.

**WARNING** automatic cherry-pick of commits failed. Commits will require human attention.

> merge conflict error: POST https://api.github.com/repos/hashicorp/consul/merges: 409 Merge conflict []

The below text is copied from the body of the original PR.

---

### Description
Updates links in our docs pointing at redirects. These are mostly links to `learn.hashicorp.com`, with a few links to `consul.io`. I've updated the paths to the correct location on developer.hashicorp.com.

---

<details>
<summary> Overview of commits </summary>

  - 1cd4fab38aa560e033f2cefeef6e12026a5fbe08  - ee6649d6d1e24218d0af505bf30ef3d0945c1183  - 46dd53475af2dfe87985674443f5aec1c053d73a 

</details>

### CODE REVIEW

**Review by github-team-consul-core-pr-approver — APPROVED** [review]
Auto approved Consul Bot automated PR

File: website/content/docs/architecture/scale.mdx [hunk 1]
```diff
@@ -196,7 +196,7 @@ In general, set [`raftboltdb.NoFreelistSync`](/consul/docs/agent/config/config-f
 - Reduce the amount of data written to disk
 - Increase the amount of time it takes to load the raft.db file on startup
 
-We recommend operators optimize networks according to their individual concerns. For example, if your server runs into disk performance issues but Consul servers do not restart often, setting [`raftboltdb.NoFreelistSync`](consul/docs/agent/config/config-files#NoFreelistSync) to `true` may solve your problems but it causes issues for deployments with large database files and frequent server restarts.
+We recommend operators optimize networks according to their individual concerns. For example, if your server runs into disk performance issues but Consul servers do not restart often, setting [`raftboltdb.NoFreelistSync`](/consul/docs/agent/config/config-files#NoFreelistSync) to `true` may solve your problems. However, the same action causes issues for deployments with large database files and frequent server restarts.
```
Comment (BRKalow) [inline website/content/docs/architecture/scale.mdx:199]: @im2nguyen Could you validate that this is the correct content here?

File: website/content/docs/architecture/scale.mdx [hunk 2]
```diff
@@ -280,4 +280,4 @@ At scale, using Consul as a backend for Vault results in increased memory and CP
...
\ No newline at end of file
+Vault v1.4 and higher provides [integrated storage](/vault/docs/concepts/integrated-storage) as its recommended storage option. If you currently use Consul as a storage backend for Vault, we recommend switching to integrated storage. For a comparison between Vault's integrated storage and Consul as a backend for Vault, refer to [storage backends in the Vault documentation](/vault/docs/configuration/storage#integrated-storage-vs-consul-as-vault-storage). For detailed guidance on migrating the Vault backend from Consul to Vault's integrated storage, refer to the [storage migration tutorial](/vault/docs/configuration/storage#integrated-storage-vs-consul-as-vault-storage). Integrated storage improves resiliency by preventing a Consul outage from also affecting Vault functionality.
```
Comment (BRKalow) [inline website/content/docs/architecture/scale.mdx:283]: @im2nguyen Could you validate that this is the correct content here?

Comment (im2nguyen) [reply website/content/docs/architecture/scale.mdx:283]: this is correct!

**Review by im2nguyen — APPROVED** [review]

File: website/content/docs/architecture/scale.mdx (hunk 1 above)
Comment (im2nguyen) [reply website/content/docs/architecture/scale.mdx:199]: also correct

**Pull request closed without merging**